lib/active_merchant/billing/gateways/fat_zebra.rb

```diff
@@ -122,11 +122,11 @@ def add_creditcard(post, creditcard, options = {})
       end
 
       def add_extra_options(post, options)
-        extra = {}
+        extra = options.fetch(:extra, {})
         extra[:ecm] = '32' if options[:recurring]
-        extra[:cavv] = options[:cavv] if options[:cavv]
-        extra[:xid] = options[:xid] if options[:xid]
-        extra[:sli] = options[:sli] if options[:sli]
+        extra[:cavv] = options[:cavv] || options.dig(:three_d_secure, :cavv) if options[:cavv] || options.dig(:three_d_secure, :cavv)
```

Comment:
The intent of this is to support both the old passthrough method and the new [standardised way of doing it](https://github.com/activemerchant/active_merchant/wiki/Standardized-3DS-Fields).